lib/urbanopt/scenario/default_reports/construction_cost.rb

```diff
@@ -35,15 +35,86 @@ module Scenario
     module DefaultReports
       class ConstructionCost 
 
+        attr_accessor :category, :item_name, :unit_cost, :cost_units, :item_quantity, :total_cost
+
         # perform initialization functions
         def initialize(hash = {})
+          hash.delete_if {|k, v| v.nil?}
+          hash = defaults.merge(hash)
+
+          @category = hash[:category]
+          @item_name = hash[:item_name]
+          @unit_cost = hash[:unit_cost]
+          @cost_units = hash[:cost_units]
+          @item_quantity = hash[:item_quantity]
+          @total_cost = hash[:total_cost]
+          
         end
         
+        def defaults
+          hash = {}
+          hash[:category] = 0
+          hash[:item_name] = 0
+          hash[:unit_cost] = 0
+          hash[:cost_units] = 0
+          hash[:item_quantity] = 0
+          hash[:total_cost] = 0
+          
+
+          return hash
+        end
+
+        
         def to_hash
           result = {}
+          result[:category] = @category if @category
+          result[:item_name] = @item_name if @item_name
+          result[:unit_cost] = @unit_cost if @unit_cost
+          result[:cost_units] = @cost_units if @cost_units
+          result[:item_quantity] = @item_quantity if @item_quantity
+          result[:total_cost] = @total_cost if @total_cost
           return result
         end
-       
+        
+        def self.merge_construction_cost(existing_cost, new_cost)
+          # modify the existing_cost by adding the :total_cost and :item_quantity
```

Comment:
@rawadelkontar this is good.  I would also add some error checking, something like:

```
if existing_cost.category != new_cost.category
  raise "Cannot merge existing cost of category #{existing_cost.category} with new cost of category #{new_cost.category}"
end
```

You can consider what the right way to handle the other fields is

Reply:
For example you might want to warn if unit costs are different?